include/readers.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef __READERS_H__
+#define __READERS_H__
+
+#include "reader_interface_classes.hpp"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+class CSVReader : public Reader {
+public:
+  CSVReader(std::string filepath) : Reader(filepath) { Read(); };
+  virtual void Read() final {
+    std::vector<std::vector<std::string>> content;
+    std::vector<std::string> row;
+    std::string line, word;
+    std::fstream file(m_filepath, std::ios::in);
+    if (file.is_open()) {
+      getline(file, line);
+      if (line == "DEFAULT") {
+        T_idx = 0;
+        rate_idx = 1;
+      }
+
+      while (getline(file, line)) {
+        row.clear();
+        std::stringstream str(line);
+        while (getline(str, word, ','))
+          row.push_back(word);
+        content.push_back(row);
+      }
+    } else {
+      std::cerr << "CSVReader: " << m_filepath << " not found" << std::endl;
+      return;
+    }
+    int nrows = content.size();
+    m_data[T_idx] = std::vector<double>(nrows);
+    m_data[rate_idx] = std::vector<double>(nrows);
+    for (int i = 0; i < content.size(); i++) {
+      try {
+        m_data[0][i] = std::stod(content[i][0]);
+        m_data[1][i] = std::stod(content[i][1]);
+      } catch (const std::invalid_argument &ia_ex) {
+        // Error handling
+        std::cerr << "CSVReader: Invalid argument converting value on line number ["
+                  << i + 2 << "] to double in file " << m_filepath << std::endl;
```

Comment:
After printing the useful info to stderr, I think we should probably rethrow the exception here - having invalid rates data is likely to be a critical error.

```suggestion
        std::cerr << "CSVReader: Invalid argument converting value on line number ["
                  << i + 2 << "] to double in file " << m_filepath << std::endl;
        throw;
```